synergy-design-system/synergy-design-system
# fix: 🤔 Vue: Using registerIconLibrary with wrappers lead to bigger bundle size

# Pull Request

## 📖 Description

This PR fixes an issue for multiple versions of synergy being loaded when importing `registerIconLibrary` from `@synergy-design-system/components`. It does this by adjusting the build step to create bundled JavaScript files from the original vue sfcs.

### 🎫 Issues

Closes #553 

## 👩‍💻 Reviewer Notes

Please have a look at the changed build step. I have opted to add a `_build` script to the `@synergy-design-system/vue` package for this to work. This also means the `vue-tsc` dependency in `@synergy-design-system/components` could be removed.

I tried using rollup directly, however this did not work as intended because of various problems with bundled peerDependencies. Therefore, `vite` was used as `rollup-plugin-vue` is deprecated for this.

The `vite` compilerOptions can now also be removed to get rid of the annoying custom-element warnings.

## 📑 Test Plan

- Check out this repository
- Install all dependencies
- Start the vue demo project
- Add `registerIconLibrary` as described in the code sample in #553
- There should no duplicate loading message.

## ✅ DoD
